protocol.md

```diff
@@ -250,15 +253,15 @@ The steps to verify the *user* using OpenID Connect are as follows:
 1. The *user* receives an `authorization code` from the OpenID Connect provider
 1. The *application* exchanges the `authorization code` to receive the *user* `ID Token` from the OpenID Connect provider
 1. The *application* provides this `ID Token` to *Tanker Core* during [user registration](#user-registration) or later on
-1. *Tanker Core* extracts the `nonce` from the `ID Token` and requests a `challenge` from the *Tanker server*
-1. The *Tanker server* generates a `challenge`
-1. The *Tanker server* records the `nonce` and the `challenge`
-1. The *Tanker server* sends the `challenge`
-1. *Tanker Core* signs the `challenge` with the secret key matching the `nonce`
-1. *Tanker Core* sends a request to the *Tanker server*. The request contains the `ID Token`, the `challenge` and the `challenge`'s signature
+1. *Tanker Core* extracts the `nonce` from the `ID Token` and requests an [OIDC Challenge] from the *Tanker server*
+1. The *Tanker server* generates an [OIDC Challenge]
+1. The *Tanker server* records the `nonce` and the [OIDC Challenge]
+1. The *Tanker server* sends the [OIDC Challenge]
+1. *Tanker Core* signs the [OIDC Challenge] with the secret key matching the `nonce`
+1. *Tanker Core* sends a request to the *Tanker server*. The request contains the `ID Token`, the [OIDC Challenge] and the [OIDC Challenge]'s signature
 1. The *Tanker server* verifies the provided `ID Token` according to [the OpenID recommendation](https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation)
-1. The *Tanker server* extracts the `nonce` from the `ID Token`, matches the `nonce`, the `challenge` and the recorded `challenge`
-1. The *Tanker server* decodes the `nonce` into a public key and verifies the `challenge`'s signature with the key
+1. The *Tanker server* extracts the `nonce` from the `ID Token`, matches the `nonce`, the request's [OIDC Challenge] and the recorded [OIDC Challenge]
+1. The *Tanker server* decodes the `nonce` into a public key and verifies the [OIDC Challenge]'s signature with the key
 1. The *Tanker server* extracts the *user*'s `subject` from the `ID Token` and hashes the `subject` before storing it.
```

Comment:
We check the ID token client somewhere right ?

Reply:
What do you mean?
> The *Tanker server* verifies the provided `ID Token` according to [the OpenID recommendation](https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation)

should cover it.

If you mean client side that's a no. We never did

Reply:
Missed that sorry!